### PR TITLE
feat: [CO-1025] [CO-1392] jdk17 support

### PR DESCRIPTION
**Pull request by sonersivri**

### CODE REVIEW

File: store-conf/pom.xml [hunk 1]
```diff
@@ -49,8 +49,6 @@
     <jar.skipIfEmpty>true</jar.skipIfEmpty>
     <maven.install.skip>true</maven.install.skip>
 
-    <maven.compiler.source>11</maven.compiler.source>
```
Comment (M0Rf30) [inline store-conf/pom.xml:52]: As "not the most competent java person": is maybe better to always explicitly set the target and source version?

**Comment by keshavbhatt** [comment]
Hi @sonersivri, The generated SOAP documentation with the new code is missing some information. When compared to the existing documentation at https://docs.zextras.com/apidoc/api-reference/index.html, a few helpful details seem to be absent. It would be really appreciated if we could bring those details back, as they are very valuable for developers.

<details><summary>Diff b/w **Existing SOAP docs** & **New SOAP docs**</summary>
<p>

Existing SOAP docs:

image


New SOAP docs:

image


</p>
</details>

**Comment by keshavbhatt** [comment]
> > Hi @sonersivri, The generated SOAP documentation with the new code is missing some information. When compared to the existing documentation at https://docs.zextras.com/apidoc/api-reference/index.html, a few helpful details seem to be absent. It would be really appreciated if we could bring those details back, as they are very valuable for developers.
> > Diff b/w **Existing SOAP docs** & **New SOAP docs**
> 
> @keshavbhatt could you check it again? thanks

@sonersivri The missing information is back :)

**Review by frisonisland — APPROVED** [review]

**Comment by keshavbhatt** [comment]
@sonersivri we should also update the Readme file of the project.  Also this line https://github.com/zextras/carbonio-mailbox/blob/feat/CO-1025-CO-1392-jdk17-support/ant-build-common.xml#L8 if it is not required anymore.

**Review by aheeva-yuliya — APPROVED** [review]

**Review by zextrasGiovanniDefacci — APPROVED** [review]

**Pull request merged**